webapi/src/main/scala/org/knora/webapi/messages/util/search/SparqlTransformer.scala

```diff
@@ -64,9 +76,14 @@ object SparqlTransformer {
 
     override def transformStatementInWhere(
       statementPattern: StatementPattern,
-      inputOrderBy: Seq[OrderCriterion]
-    ): Seq[StatementPattern] =
-      transformStatementInWhereForNoInference(statementPattern = statementPattern, simulateInference = true)
+      inputOrderBy: Seq[OrderCriterion],
+      limitInferenceToOntologies: Option[Set[SmartIri]] = None
+    )(implicit executionContext: ExecutionContext): Seq[QueryPattern] =
```

Comment:
I don't really understand why you added the execution context here and in all other cases and why it worked before without. Maybe you can explain that to me once?

Reply:
previously the SparqlTransformer did not use any futures, so it didn't need any execution context. Now for my optimization I do use futures, so I do need the execution context.  
There could probably be more elegant solutions - but once we use ZIO, we'll be using less futures, so this will change anyways, that's why I didn't try to be cosmetic about it.

Reply:
Also, execution context is noted as a knowledge transfer topic, which I really need too ^^